tests/test_templates_broadcast.py

```diff
@@ -65,39 +62,52 @@ def KwargTemplateDouble(par, wires, a=True):
+                  ("double", 4, [pi / 2, pi / 2], CRX, [-1, 0, -1, 0]),
+                  ("double", 4, None, CNOT, [-1, 1, -1, 1]),
+                  ("double_odd", 4, [pi / 2], CRX, [-1, -1, 0, -1]),
+                  ("chain", 4, [pi, pi, pi / 2], CRX, [-1, 1, -1, 0]),
+                  ("ring", 4, [pi, pi, pi / 2, pi], CRX, [0, 1, -1, 0]),
+                  ("pyramid", 4, [0, pi, pi / 2], CRX, [-1, -1, 0, 1]),
+                  ("all_to_all", 4, [pi/2, pi/2, pi/2, pi/2, pi/2, pi/2], CRX, [-1, 0, 1/2, 3/4])
                   ]
 
-CV_TARGET_OUTPUTS = [("single", [[0.1, 0.0], [0.2, 0.0], [0.3, 0.0], [0.4, 0.0]], Displacement, [2.2, 2.4, 2.6, 2.8],
-                      qml.X),
-                     ("double", [[pi / 4, 0.0], [pi / 4, 0.0]], Beamsplitter, [0, 2, 0, 2],
-                      qml.NumberOperator),
-                     ]
-
-GATE_PARAMETERS = [("single", RX, [[0.1], [0.2], [0.3]]),
-                   ("single", Rot, [[0.1, 0.2, 0.3], [0.3, 0.2, 0.1], [0.3, 0.2, -0.1]]),
-                   ("single", T, [[], [], []]),
-                   ("double", CRX, [[0.1]]),
-                   ("double", CRot, [[0.1, 0.2, 0.3]]),
-                   ("double", CNOT, [[]]),
-                   ("double_odd", CRX, [[0.1]]),
-                   ("double_odd", CRot, [[0.3, 0.2, 0.1]]),
-                   ("double_odd", CNOT, [[]]),
+GATE_PARAMETERS = [("single", 3, RX, [[0.1], [0.2], [0.3]]),
+                   ("single", 3, Rot, [[0.1, 0.2, 0.3], [0.3, 0.2, 0.1], [0.3, 0.2, -0.1]]),
+                   ("single", 3, T, [[], [], []]),
+                   ("double", 3, CRX, [[0.1]]),
+                   ("double", 3, CRot, [[0.1, 0.2, 0.3]]),
+                   ("double", 3, CNOT, [[]]),
+                   ("double_odd", 3, CRX, [[0.1]]),
+                   ("double_odd", 3, CRot, [[0.3, 0.2, 0.1]]),
+                   ("double_odd", 3, CNOT, [[]]),
+                   ("chain", 3, CRX, [[0.1], [0.1]]),
+                   ("chain", 3, CRot, [[0.3, 0.2, 0.1], [0.3, 0.2, 0.1]]),
+                   ("chain", 3, CNOT, [[], []]),
+                   ("ring", 3, CRX, [[0.1], [0.1], [0.1]]),
+                   ("ring", 3, CRot, [[0.3, 0.2, 0.1], [0.3, 0.2, 0.1], [0.3, 0.2, 0.1]]),
+                   ("ring", 3, CNOT, [[], [], []]),
+                   ("pyramid", 3, CRX, [[0.1]]),
+                   ("pyramid", 4, CRX, [[0.1], [0.1], [0.1]]),
+                   ("pyramid", 4, CRot, [[0.3, 0.2, 0.1], [0.3, 0.2, 0.1], [0.3, 0.2, 0.1]]),
+                   ("pyramid", 4, CNOT, [[], [], []]),
+                   ("all_to_all", 3, CRX, [[0.1], [0.1], [0.1]]),
+                   ("all_to_all", 4, CRX, [[0.1], [0.1], [0.1], [0.1], [0.1], [0.1]]),
+                   ("all_to_all", 4, CRot, [[0.3, 0.2, 0.1], [0.3, 0.2, 0.1], [0.3, 0.2, 0.1],
+                                            [0.3, 0.2, 0.1], [0.3, 0.2, 0.1], [0.3, 0.2, 0.1]]),
+                   ("all_to_all", 4, CNOT, [[], [], [], [], [], []]),
```

Comment:
Cool!

Reply:
I improved the tests a bit: Instead of always testing patterns, now different patterns are only tested when checking the target state prepared, and when checking the parameter queue. I added test cases for 0, 1, 2 wires for all patterns, because these contain a lot of exceptions...